### ISSUE TITLE
Synchronous tasks?

### ISSUE THREAD

**Issue by billerr** [catdad/gulp-multistream#2]
Hi! We're using this plugin in a task that needs to run as a dependency, in the context of a watch task with browserSync integration. We have a structure similar to the following:
```js
gulp.task( 'task1', function(){
  ...
  gulp.watch('glob', ['task2', 'task3']);
});

gulp.task('task2', function() {
  return gulp.src('glob')
    .pipe(...)
    .pipe(multistream(
      gulp.dest('dirpath1'),
      gulp.dest('dirpath2')
    ));
});

gulp.task('task3', ['task2'], function() {
  browserSync.reload();
});
```

This setup should ensure that *task2* is completed (and all files in their place) before *task3* reloads the browsers (caching is disabled). However, almost always the reload happens before the files are in place, and we need a second, manual refresh to get them to load. 

Do the multistream operations run concurrently, or is there something else wrong in this setup and the tasks run asynchronously?

**Comment by catdad** [comment]
Hmm... interesting. This was always a bit of a hack, but you are right, there is a bug in `gulp-multistream`. Currently, it waits to **read** all of the files, but it does not wait for all the destination streams to flush before it is done. This is kind of a problem, as you have noticed.

I already have a fix, but this project has bad tests, so I am adding a couple of tests before publishing. Sit tight.

**Comment by catdad** [comment]
A fix for this bug was published in v0.1.1. The multistream will now wait for all destination streams to be flushed and end before ending itself (and therefore, before the gulp task ends), so your tasks should execute in order now.

Also, in your watch task, you can leave `task2` out of the list. Right now, that task will run twice, and there's really no need for that.

Let me know if you have any more trouble.

**Comment by billerr** [comment]
Great! Thanks for the prompt response!
I tested a bit and it works great now, it's been pushed to the rest of the team.
Thank you!
